### ISSUE TITLE
Battle Camera doesn't work right in at least some cases

### ISSUE THREAD

**Issue by Trivial-Man** [Albeoris/Memoria#222]
The initial camerawork at the start of a battle will sometimes focus on seemingly random and empty bits of scenery instead of the enemies. This can be seen very clearly by starting a battle with Adamantoise and seeing the camera dramatically zoom and twist just to the left of him instead of on his face like the PSX version does. This seems to be an issue with the remake and not with mods, so fixing it would be an enhancement.

I don't know where else this happens, maybe this is an isolated incident, but this particular enemy has a few issues that all together made me take notice

**Comment by Tirlititi** [comment]
It is not an isolated incident either.
(1) Battle camera movements are a bit off, it concerns all the battles. It was widely noticed in Beatrix battles for instance.
(2) There are a couple of camera movements that are simply not performed in the PC version while they existed in the PSX version. The camera movements when Ragtime Mouse reads its question and when it starts reacting is an example of that (eg. at 0:45 of [this video](https://www.youtube.com/watch?v=lbN5uQX0vlM), to be compared with 0:45 of [that one](https://www.youtube.com/watch?v=IoH1fM4fk3E)).
(3) Some camera movements do not focus the correct target. Again, the Ragtime Mouse shows an example of that: the starting camera should focus first O then X, but in the PC version it focuses first Ragtime then X. It might be that the Adamantoise issue is that one, but I rather think it's the (1).
(4) Also, and this is due to the SFX rework I did in Memoria, so it's not a problem of the PC port contrary to the others, the camera movements at the start of spell casts are not performed.

About (1), I am not sure exactly of what is the problem (it could be that the whole world is shifted a bit, or that the PSX camera matrix is slightly wrongly interpretated and converted into an Unity camera...). I guess the effect could be fixed or lessened with trials and errors. However, it could only be fixed with the engaging camera movements because it's not currently possible to have control over cameras at times when spell special effects are rendered.

About the others, I have no idea of how to fix them as long as both the camera movement file formats and the SFX file formats are not deciphered, which is a very difficult task.

**Comment by snouz** [comment]
@Tirlititi Is this still a bug? Didn't you rework the whole SFX since this ticket was opened?

**Comment by Tirlititi** [comment]
Yes, there are still multiple bugs concerning battle cameras and the state of these bugs have not really changed since this ticked was opened.
All these bugs, (1), (2), (3) and (4), still exist and I still think they will be very difficult to overcome.

**Comment by yisandind** [comment]
https://github.com/user-attachments/assets/f60beaf7-9581-4e73-9c50-a67ba95f7fec

maybe if the command menu isn't finished (Zidane's command menu is still available), the magic animation camera works fine.


https://github.com/user-attachments/assets/e0af4995-0b57-4804-9603-c583cc4d2b77

then maybe if the command menu is complete (command menu not available, all party members have command queue active), the magic animation camera doesn't work properly.

I repeated it many times it remains like that. any ideas?
